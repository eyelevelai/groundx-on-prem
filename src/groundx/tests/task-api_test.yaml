suite: "app renders as expected"
values:
  - ../values.yaml
chart:
  appVersion: 0.1.0
  version: 0.1.0

tests:
  - it: task-api.yaml
    templates:
      - ../templates/app/task-api.yaml
    asserts:
      - matchSnapshot: {}
