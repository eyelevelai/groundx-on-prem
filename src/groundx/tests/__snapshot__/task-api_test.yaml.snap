task-api.yaml:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: layout-api
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: layout-api
      namespace: eyelevel
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: layout-api
      template:
        metadata:
          labels:
            app: layout-api
        spec:
          containers:
            - command:
                - /bin/bash
                - -c
                - |
                  export PYTHONPATH=/app && LOCAL=0 gunicorn -c /app/gunicorn_conf.py --capture-output --enable-stdio-inheritance
              env:
                - name: POD_NAME
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.name
              image: public.ecr.aws/c9r4x6y5/eyelevel/python-api:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health
                  port: 8080
                initialDelaySeconds: 30
                periodSeconds: 30
              name: layout-api
              ports:
                - containerPort: 8080
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /health
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 30
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                fsGroup: 1001
                fsGroupChangePolicy: OnRootMismatch
                runAsGroup: 1001
                runAsNonRoot: true
                runAsUser: 1001
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /app/config.py
                  name: config-volume
                  subPath: config.py
                - mountPath: /app/gunicorn_conf.py
                  name: gunicorn-conf-volume
                  subPath: gunicorn_conf.py
              workingDir: /app
          initContainers:
            - command:
                - sh
                - -c
                - until nc -z cache.eyelevel.svc.cluster.local 6379; do echo waiting for cache; sleep 2; done
              image: public.ecr.aws/c9r4x6y5/eyelevel/busybox:latest
              imagePullPolicy: Always
              name: wait-for-cache
          nodeSelector:
            node: eyelevel-cpu-only
          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            fsGroup: 1001
            fsGroupChangePolicy: OnRootMismatch
            runAsGroup: 1001
            runAsNonRoot: true
            runAsUser: 1001
            seccompProfile:
              type: RuntimeDefault
          tolerations:
            - effect: NoSchedule
              key: node
              value: eyelevel-cpu-only
          volumes:
            - configMap:
                name: layout-config-py-map
              name: config-volume
            - configMap:
                name: layout-gunicorn-conf-py-map
              name: gunicorn-conf-volume
